kubernetes/config/incluster_config.c

```diff
@@ -47,7 +47,7 @@ static int setBasePathInCluster(char **pBasePath)
 
 #if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__APPLE__)
     const char *service_host_env = getenv(SERVICE_HOST_ENV_NAME);
-#elif __linux
+#elif __linux || defined(__EMSCRIPTEN__)
```

Comment:
Which platform do you test on ?  Windows ?

Reply:
Tested on Ubuntu and Windows. Compiled using latest emsdk.

Reply:
On Ubuntu,  it's enough to use `#elif __linux` to enable `secure_getenv`

On Windows, what's the precompiled result ？
```
#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__APPLE__)
    const char *service_host_env = getenv(SERVICE_HOST_ENV_NAME);
```
also takes effect.

Reply:
I think that there is a misunderstanding here.
These flags refer to the target OS, not the host where compilation is taking place.
This fix is for targeting WASM32 when compiled using Emscripten. The resulting binary neither runs on Windows nor Linux. It runs inside the browser. __linux and WIN32 are not set when targeting Emscripten. 
https://en.wikipedia.org/wiki/WebAssembly

Reply:
The #if preprocessor directive takes effect in the compilation process.

But I get it that Emscripten/WASM is neither Windows nor Linux.

The change is OK to me.